src/upgrade-db.js

```diff
@@ -0,0 +1,274 @@
+
+        this.dbState = new SMTTmpDb(this.rollupDb.db);
+        this.stateTree = new SMT(this.dbState, this.rollupDb.stateRoot);
+        this.currentBatch = this.rollupDb.lastBatch + 1;
+    }
+
+    /**
+     * Perform upgrade for each account to new leaf state
+     */
+    async doUpgrade() {
+        let initKeyStateTree;
+        let finalKeyStateTree;
+
+        for (let i = 0; i < this.numProofs; i++) {
+            initKeyStateTree = this.initIdx + i * this.nAccounts;
+            finalKeyStateTree = this.initIdx + (i + 1) * this.nAccounts;
+            await this._buildSingleProof(initKeyStateTree, finalKeyStateTree);
+        }
+
+        const insertsState = Object.keys(this.dbState.inserts).reverse().map((key) => {
+            return [Scalar.e(key), this.dbState.inserts[key]];
+        });
+
+        await this.rollupDb.db.multiIns([
+            ...insertsState,
+            [Scalar.add(Constants.DB_Batch, this.currentBatch), this.stateTree.root],
+            [Scalar.add(Constants.DB_InitialIdx, this.currentBatch), this.maxIdx],
+            [Constants.DB_Master, this.currentBatch]
+        ]);
+
+        const newRollupDB = await RollupDB(this.rollupDb.db, this.rollupDb.chainID, this.currentBatch - 1);
+        this.isUpgraded = true;
+
+        return newRollupDB;
+    }
+
+    /**
+     * Build single proof for upgrading accounts
+     * @param {Number} initKey - first key to update
+     * @param {Number} finalKey - last key to update
+     */
+    async _buildSingleProof(initKey, finalKey) {
+        const input = this._initInput();
+        const imStateRoot = [];
+        const imFinalIdx = [];
+        this.finalIdxs.push((finalKey < this.maxIdx) ? finalKey : this.maxIdx + 1);
+
+        input.initialIdx = initKey;
+        input.maxIdx = this.maxIdx;
+        input.oldStateRoot = this.stateTree.root;
+
+        for (let i = initKey; i < finalKey; i++) {
+            const k = i - initKey;
+
+            if (i <= this.maxIdx) {
+                // get old state
+                let oldState;
+                const resFind = await this.stateTree.find(i);
+                if (resFind.found) {
+                    const foundValueId = poseidonHash([resFind.foundValue, i]);
+                    oldState = commonjsOld.stateUtils.array2State(await this.dbState.get(foundValueId));
+                } else {
+                    throw new Error(`Unreacheable code: key state tree ${i} does not exist`);
+                }
+
+                // set new state
+                const newState = Object.assign({}, oldState);
+                newState.exitBalance = 0;
+                newState.accumulatedHash = 0;
+                const newValue = stateUtils.hashState(newState);
+
+                const res = await this.stateTree.update(i, newValue);
+                let siblings = res.siblings;
+                while (siblings.length < this.nLevels + 1) siblings.push(Scalar.e(0));
+
+                // Database Idx
```

Comment:
There are more databases to upgrade, such as:
- https://github.com/hermeznetwork/commonjs/blob/master/src/batch-builder.js#L429
- https://github.com/hermeznetwork/commonjs/blob/master/src/batch-builder.js#L468

Reply:
also, there were deleted in the upgrade-v1 https://github.com/hermeznetwork/commonjs/blob/feature/upgrade-v1/src/batch-builder.js. I understand because they are usefull when an account is created and  not updated, therefore, there's no need to add it here right?

Reply:
Although you are right regarding the update of the accounts. It nos not necessary to add idx values to sign-Ay or ethAddr since we are just updating the accounts, not adding new ones.